src/modules/cmdpal/Exts/Microsoft.CmdPal.Ext.WindowWalker/Helpers/VirtualDesktopHelper.cs

```diff
@@ -0,0 +1,543 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Text;
+
+using Microsoft.CmdPal.Ext.WindowWalker.Properties;
+using Microsoft.CmdPal.Extensions.Helpers;
+using Microsoft.Win32;
+
+namespace Microsoft.CmdPal.Ext.WindowWalker.Helpers;
+
+/// <summary>
+/// Helper class to work with Virtual Desktops.
+/// This helper uses only public available and documented COM-Interfaces or information from registry.
+/// </summary>
+/// <remarks>
+/// To use this helper you have to create an instance of it and access the method via the helper instance.
+/// We are only allowed to use public documented com interfaces.
+/// </remarks>
+/// <SeeAlso href="https://learn.microsoft.com/windows/win32/api/shobjidl_core/nn-shobjidl_core-ivirtualdesktopmanager">Documentation of IVirtualDesktopManager interface</SeeAlso>
+/// <SeeAlso href="https://learn.microsoft.com/archive/blogs/winsdk/virtual-desktop-switching-in-windows-10">CSharp example code for IVirtualDesktopManager</SeeAlso>
+public class VirtualDesktopHelper
```

Comment:
ooh I need this for the Virtual desktop switcher. Heck we should just make them one builtin with two different top-levels. 

(don't do that as a part of this PR)

Reply:
Yeah one builtin with two different top levels makes a lot of sense - I like it.